iceoryx_posh/include/iceoryx_posh/popo/untyped_publisher.hpp

```diff
@@ -38,22 +42,24 @@ class UntypedPublisherImpl : public base_publisher_t
     virtual ~UntypedPublisherImpl() = default;
 
     ///
-    /// @brief loan Get a chunk from loaned shared memory.
-    /// @param size The expected size of the chunk.
+    /// @brief Get a chunk from loaned shared memory.
+    /// @param payloadSize The expected user payload size of the chunk.
+    /// @param payloadAlignment The expected user payload alignment of the chunk.
     /// @return A pointer to a chunk of memory with the requested size or
     ///         an AllocationError if no chunk could be loaned.
     /// @note An AllocationError occurs if no chunk is available in the shared memory.
     ///
-    cxx::expected<void*, AllocationError> loan(const uint32_t size) noexcept;
+    cxx::expected<void*, AllocationError> loan(const uint32_t payloadSize,
+                                               const uint32_t payloadAlignment = 8U) noexcept;
```

Comment:
Is the `8` not defined in some constexpr value already?

Reply:
This was a placeholder in the beginning since I was not sure if this should be an optional parameter. I kept it that way but I'm still not sure if it should be mandatory to pass the payload alignment

Reply:
If we do not have a default alignment constant which we use elsewhere we can keep the 8U here. I find an optional argument ok but in the C API we will have to pass both. The other question is what is a reasonable default value? 8U targets 64 Bit architectures I guess and is fine.